packages/oauth/src/types/oauth-options.ts

```diff
@@ -2,6 +2,13 @@ import { OAuthUser } from './oauth-user';
 
 export interface OAuthOptions {
   [provider: string]: {
+    /**
+     * Should return oauth user payload from passed authentication code or token
+     */
     authenticate: (params: any) => Promise<OAuthUser>;
+    /**
+     * Allows to provide function to create custom payload for user creation method based on data from OAuth API
+     */
+    getRegistrationPayload?: (oauthUser: OAuthUser) => Promise<any>;
```

Comment:
can we pass the provider name as a first param? that way you would be able to do some conditional logic based on twitter of facebook for example

Reply:
but we already write it inside `provider` object https://github.com/accounts-js/accounts/pull/920/files/900157dedadcb7e479e8379dad1799e9fc8364ac#diff-dc3049153df06cd271b8ee347f5852e8R4

wrote more details about this in this comment: https://github.com/accounts-js/accounts/pull/920#issuecomment-610845227

Reply:
true sorry I totally missed the line 4 so I thought it was directly passed as an option 👍